data-prepper-api/src/main/java/org/opensearch/dataprepper/model/types/ByteCount.java

```diff
@@ -120,4 +121,48 @@ private static BigDecimal scaleToBytes(final BigDecimal value, final Unit unit)
     private static boolean isFractional(final BigDecimal value) {
         return value.remainder(BigDecimal.ONE).compareTo(BigDecimal.ZERO) != 0;
     }
+
+    /**
+     * Returns a hash code consistent with {@link ByteCount#equals(Object)}.
+     *
+     * @return The hash code
+     * @since 2.7
+     */
+    @Override
+    public int hashCode() {
+        return Objects.hashCode(bytes);
+    }
+
+    /**
+     * Returns whether this and another instance are equal {@link ByteCount} values.
+     * Two objects are equal if the bytes are equal. It does not matter how they
+     * were originally parsed.
+     *
+     * @param otherObject The other object to compare against.
+     * @return True if and only if the values are equal.
+     * @since 2.7
+     */
+    @Override
+    public boolean equals(final Object otherObject) {
+        if(! (otherObject instanceof ByteCount))
+            return false;
+
+        return ((ByteCount) otherObject).bytes == bytes;
+    }
+
+    /**
+     * Returns a string representation of this {@link ByteCount}. This
+     * representation can be parsed by {@link ByteCount#parse(String)}.
+     * <p>
+     * The exact string return could change between versions, but this
+     * API will provide consistency that the value is accurate and can
+     * be parsed by {@link ByteCount#parse(String)}.
+     *
+     * @return The string representation of this byte count.
+     * @since 2.7
+     */
+    @Override
+    public String toString() {
+        return bytes + "b";
```

Comment:
Nit: magic string. Took me a bit to realize the meaning of `b`.

Reply:
Thanks! I corrected it in this PR: https://github.com/opensearch-project/data-prepper/pull/3993